### PR TITLE
upcoming: [M3-8840] - Add Cluster Type selection to LKE Create Cluster flow

### PR DESCRIPTION
**Pull request by mjac0bs**
## Description 📝

We want to allow users the ability to create an LKE-Enterprise cluster in the create flow. 

## Changes  🔄

- A new Cluster Type section exists in the Create Cluster form.
- Copy describes the section and a docs link with placeholder content.
- The cards handle enabled/disabled state as follows:
   - When the feature flag is disabled, do not display the Cluster Type Panel.
   - When the feature flag is enabled but the user lacks the LKE-Enterprise account capability, the Enterprise panel is disabled and displays a tooltip.
   - When the feature flag is enabled and the user has the LKE-Enterprise account capability, the Enterprise panel is enabled and LKE (standard) is selected by default. 
- When the Enterprise cluster type is enabled, the HA Control Plane section is not visible.
- Test coverage confirms the above.

## Preview 📷

| Flag ON with Capability  | Flag ON without Capability  |
| ------- | ------- |
| Screenshot 2024-11-25 at 1 52 27 PM Screenshot 2024-11-25 at 1 52 34 PM | Screenshot 2024-11-25 at 1 34 28 PM Screenshot 2024-11-25 at 1 34 44 PM |

### Prerequisites

(How to setup test environment)

- Have LKE-E customer tag on your account (see Project Tracker)

### Verification steps

(How to verify changes)
**Note: an actual LKE-E cluster is not creatable yet. An LKE-E version will need to be set, which is an upcoming ticket (M3-8848).**

- [ ] Go to http://localhost:3000/kubernetes/create
- [ ] Confirm the Cluster Type section shows as expected:
  - [ ] Feature flag in dev tools ON, customer tag ON: Cluster Type is visible; HA section is not visible when Enterprise is selected
   - [ ] Feature flag in dev tools OFF, customer tag ON: Cluster Type is not visible; HA section is always visible
   - [ ] Feature flag in dev tools ON, customer tag OFF: Cluster Type is visible; Enterprise selection is disabled; HA section is always visible
- [ ] Confirm test coverage passes:
```
yarn cy:run -s "cypress/e2e/core/kubernetes/lke-create.spec.ts"
```

## As an Author, I have considered 🤔

- 👀 Doing a self review
- ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- 🤏 Splitting feature into small PRs
- ➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- 🧪 Providing/improving test coverage
- 🔐 Removing all sensitive information from the code and PR description
- 🚩 Using a feature flag to protect the release
- 👣 Providing comprehensive reproduction steps
- 📑 Providing or updating our documentation
- 🕛 Scheduling a pair reviewing session
- 📱 Providing mobile support
- ♿  Providing accessibility support

<br/>

- [x] I have read and considered all applicable items listed above.

## As an Author, before moving this PR from Draft to Open, I confirmed ✅

- [x] All unit tests are passing
- [x] TypeScript compilation succeeded without errors
- [x] Code passes all linting rules

### CODE REVIEW

File: packages/manager/src/components/SelectionCard/SelectionCard.tsx [hunk 1]
```diff
@@ -87,6 +88,11 @@ export interface SelectionCardProps {
    * Optional text to set in a tooltip when hovering over the card.
    */
   tooltip?: JSX.Element | string;
+  /**
```
Comment (mjac0bs) [inline packages/manager/src/components/SelectionCard/SelectionCard.tsx:91]: Added this prop to be able to control the placement of the tooltip - it was covering the helper text and we could avoid that by placing elsewhere.

File: packages/manager/src/features/Kubernetes/CreateCluster/CreateCluster.styles.ts [hunk 2]
```diff
@@ -51,11 +51,10 @@ export const StyledFieldWithDocsStack = styled(Stack, {
 }));
 
 export const StyledDocsLinkContainer = styled(Box, {
-  label: 'StyledRegionSelectStack',
+  label: 'StyledDocsLinkContainer',
```
Comment (mjac0bs) [inline packages/manager/src/features/Kubernetes/CreateCluster/CreateCluster.styles.ts:54]: Changes to use the same styling for both the cluster type and DC pricing docs links.

File: packages/manager/src/features/Kubernetes/kubeUtils.ts [hunk 3]
```diff
@@ -191,29 +191,34 @@ export const getLatestVersion = (
  * Hook to determine if the LKE-Enterprise feature should be visible to the user.
  * Based on the user's account capability and the feature flag.
  *
- * @returns {boolean, boolean} - Whether the LKE-Enterprise feature is enabled for the current user in LA and GA, respectively.
+ * @returns {boolean, boolean, boolean, boolean} - Whether the LKE-Enterprise flags are enabled for LA/GA and whether feature is enabled for LA/GA (flags + account capability).
```
Comment (mjac0bs) [inline packages/manager/src/features/Kubernetes/kubeUtils.ts:194]: Changes were to account for the fact that the Create flow shows the Cluster Type section as long as the LD feature flag is on; it does *not* depend on account capability. The reason for this is to show all customers what is potentially available, even if not currently on their contract.

**Review by hana-akamai — COMMENTED** [review]
I think we'll need to explicitly clear the HA values and/or cluster type when LKE Enterprise is selected. Right now, if we go through the flow for LKE and select HA, then switch to LKE-E at the end, what ends up getting created is LKE HA instead of LKE-E

**Review by hana-akamai — APPROVED** [review]

**Review by cpathipa — APPROVED** [review]
Nice work!  The functionality is working as expected, confirming the behavior of the Cluster Type section and HA section when toggling the feature flag and account tag.

image

**Pull request merged**